### PR TITLE
Upstream DynamoDbExceptionMapper

### PR DESCRIPTION
**Pull request by adrw**
This has proved to be a useful set of classes for services that organically encounter these DynamoDb exceptions solved by transaction retry, and results in less exceptions and in correctly mapped 503 responses.

### CODE REVIEW

File: misk-aws2-dynamodb/build.gradle [hunk 1]
```diff
@@ -3,6 +3,7 @@ dependencies {
 
   implementation dep.guice
   implementation project(':misk-aws')
+  implementation project(':misk-exceptions-dynamodb')
```
Comment (adrw) [inline misk-aws2-dynamodb/build.gradle:6]: I tried to find these exceptions in the aws2 SDK and couldn't find which Gradle dependency to include. Any ideas?

File: misk-aws-dynamodb/build.gradle [hunk 2]
```diff
@@ -2,6 +2,7 @@ dependencies {
   implementation dep.guice
   implementation dep.awsDynamodb
   implementation project(':misk-aws')
+  implementation project(':misk-exceptions-dynamodb')
```
Comment (chris-ryan-square) [inline misk-aws-dynamodb/build.gradle:5]: Maybe switch this to "api" instead of "implementation"?

**Review by zhxnlai — APPROVED** [review]

**Comment by adrw** [comment]
@chris-ryan-square Have you seen the failing tests before? Seem related to RepeatedTaskQueueTest and unrelated to this PR.

**Comment by chris-ryan-square** [comment]
@adrw could you bring this PR up to date please?

**Comment by adrw** [comment]
Sorry, having a hard time getting my local Misk repo in working shape, will take another stab at it this week

**Pull request merged**